apps/api/src/modules/payments/payments.routes.ts

```diff
@@ -23,7 +24,12 @@ export async function getPublicKey(
   const paymentService = request
     .getContainer()
     .get<PaymentsService>(PaymentsService.name)
-  reply.send(await paymentService.getPublicKey())
+  const publicKey = await paymentService.getPublicKey()
+  if (publicKey) {
+    reply.send(publicKey)
+  } else {
+    reply.notFound()
+  }
```

Comment:
This fixes the issue where the public key endpoint was failing but it showed as successful